### DISCUSSION TITLE
TailwindCSS vs Plain CSS

### DISCUSSION THREAD

**Discussion by azzamsa** [welpo/tabi#242] [Q&A]
Hi, Óscar!

I plan to migrate my previous (big) blog to Zola.
I have never built a large site using plain CSS/SASS; I have used TailwindCSS for all my 'production-grade' projects.
However, now I am tempted to decide whether to go with a fully single binary, Zola + SASS/Bluma, or to stick with TailwindCSS.

I am curious about your thoughts. Is there any reason you chose 'plain' CSS/SASS compared to something like TailwindCSS?

Thank you for Tabi! 🎖️

**Comment by welpo** [comment]
Hello, Azzam!

The honest truth is tabi was never meant to be what it is now. It started with "let's just modify this theme ([shadharon](https://github.com/syedzayyan/shadharon)) a bit". And here we are.

At this point, I see no compelling reason to use TailwindCSS. Plus, I'm not a fan of the class soup.

I've always been a fan of minimal setups and DIY (perhaps to a fault); I love that my site can be set up with a single binary (Zola), with no extra dependencies. This minimalism is what led me to Zola.

Hope that answers your question! Happy to discuss this.

**Comment by azzamsa** [comment]
> At this point, I see no compelling reason to use TailwindCSS. Plus, I'm not a fan of the class soup.

I don't have any bold reason for choosing 'why.' I think it is solely because I started everything with TailwindCSS, and I don't have any experience building a site with plain CSS.

> I've always been a fan of minimal setups and DIY (perhaps to a fault); I love that my site can be set up with a single binary (Zola), with no extra dependencies. This minimalism is what led me to Zola.

This is the reason I came here. I used 11ty for the past 3 years for my personal website. But I always wanted something that is 'set and forget.' Unlike Rust apps, in my personal experience, JavaScript and Python projects are ones where if you leave them sitting for 3 years, then you have to upgrade the dependencies. There is no guarantee that it will work, even installing the old dependencies is already a problem. So, there is a constant fear of 'If I don't do regular maintenance of my site, it will break someday (when the dependencies are too old).'

But coming to Zola, I don't know if I am able to replicate my 'not simple' blog in plain CSS. Currently, I am looking at BulmaCSS and PicoCSS.

What do you think?

**Comment by welpo** [comment]
If you already have a working site, why not keep the CSS? Less work :)

I have 0 experience with Tailwind, Bulma or Pico, so I can't really help there, sorry.

**Comment by Jieiku** [comment]
Most Zola themes use either custom css or a css/sass framework like the ones you listed, some of them even use bits and pieces from multiple css frameworks.

I would say that if you are trying to accomplish features that do not exist in a particular Zola theme then you will need to be up to learning not only some css/sass but also the tera templating language that Zola uses. Even if you use an existing sass framework, you still have to learn how to use that framework. Some frameworks make heavy use of classes, which means you have to learn how those classes work, while other frameworks have rules for semantic html, possibly with a small subset of utility classes where it makes sense, meaning all you really have to know is html.

My previous blog used Hugo, and I relied solely on the features and appearance of the theme that I was using at that time. I had created many websites before and even though I was comfortable with HTML I never really created anything complex from scratch.

I created my own theme because none of the themes did everything that I was looking for, and there are still some todo items on my list but I am comfortable with what I have now. It took me about 6 months or so to create something that works for me, and I have steadily improved it to make it more versatile for multiple use cases.

I never used tailwind or bulma because I found them to be a bit heavy, I have however used Pico for a few months and almost stuck with that before writing my own css framework that is based on bamboo css, which in turn was based on new.css. I did incorporate a few similar features that were in Pico as well.

So my answer or opinion is that if there is a Zola theme that has all or most of the features your looking for then you won't have a problem, some of the themes may even use the css/sass frameworks you listed. If you don't mind taking some time to learn css/sass and the tera templating language then you will also be fine.

I feel I am pretty familiar with Zola, and will say that maybe its only shortcoming is that you cannot template a javascript file, which can limit or complicate accomplishing certain features, in my theme I have used npm and javascript to overcome some of Zola's shortcomings and am still happy to be using Zola because it is pretty straightforward to use once you learn it.

Luckily there are a LOT of Zola themes, so you can usually look at how a feature was accomplished in one theme and add it to another theme.

I hope this info is helpful for your decision.
